app/controllers/concerns/base_application_controller_behavior.rb

```diff
@@ -0,0 +1,117 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+require 'application_responder'
+
+module BaseApplicationControllerBehavior
+  extend ActiveSupport::Concern
```

Comment:
Can we chat about this changes @ttoomey with @eanders?
I'm vaguely afraid of having two application controllers (or rather, two controllers inheriting from `< ActionController::Base`) for a single application, but that might just be my unfamiliarity with Rails.

A couple thoughts:
- No matter how its set up, I don't think we should duplicate all this logic. @ttoomey were you thinking that the warehouse ApplicationController would also use this `BaseApplicationControllerBehavior` concern? 
- I'm unclear on how it was decided which parts of ApplicationController are needed for HMIS and which are not. For example `bypass_2fa_enabled?` didn't get moved to the concern, but in HMIS we have a controller that uses a concern that expects that to be present, so is that going to break now? That's just one example, but overall I would be concerned about this leading to problems using shared concerns that expect the ApplicationController to be consistent

Reply:
Yes, I'd be happy to huddle about it! I figured this would need some discussion 

> No matter how its set up, I don't think we should duplicate all this logic.

I considered extracting the common behavior but was concerned about the risk of modifying ApplicationController. The HMIS side is a much smaller surface to test. Happy to take another pass at that and DRY it up but I want to confirm that's a direction we should go.

> I'm unclear on how it was decided which parts of ApplicationController are needed for HMIS and which are not.

Yeah, this definitely needs thought and QA. Sorry I missed `bypass_2fa_enabled` - that was an oversight on my part. There is definitely a risk with this change.

An alternative to this approach is that we create a new BaseGrdaWarehouseController that inherits from the existing ApplicationController but includes the devise/pretender config. Then we change the current occurrences of ApplicationController to inherit from BaseGrdaWarehouseController instead (looks like about 350 occurrences). That may bring less risk but it does result in code churn

Reply:
Thanks for the explanations @ttoomey!  Given all that, I think I am on-board with moving forward with a second controller for HMIS. I agree it feels risky to modify the warehouse ApplicationController, and it will be much easier to just test the HMIS. What do you think @eanders?

Reply:
Yeah, feels like the right approach.  I wouldn't be opposed to a concern like this, as long as what's in here follows the current application controller.